src/Columns/ColumnFunction.cpp

```diff
@@ -308,6 +308,7 @@ ColumnWithTypeAndName ColumnFunction::reduce() const
         ProfileEvents::increment(ProfileEvents::CompiledFunctionExecute);
 
     res.column = function->execute(columns, res.type, elements_size);
+    chassert(res.column->getDataType() == res.type->getTypeId());
```

Comment:
This is too optimistic. I don't remember exactly what column->getDataType returns, but there is no 1:1 correspondence between columns and data types. For example, ColumnUInt16 can correspond to UInt16 or Date or Enum16.

Reply:
Let's see if I can find something similar. This would help the fuzzer, as it would trigger without input data.

Reply:
I'm going to move the asserts to a different PR so it's easier to backport.